src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js

```diff
@@ -63,64 +63,64 @@
-                var headerChanged = Boolean(templateMetadataView.pageHeader !== templateMetadata.pageHeader);
-
-                templateMetadataView.labels = templateMetadata.labels;
-                templateMetadataView.mode = templateMetadata.mode;
-                templateMetadataView.hasError = templateMetadata.hasError;
-                templateMetadataView.errorMessage = templateMetadata.errorMessage;
-                templateMetadataView.pageHeader = templateMetadata.pageHeader;
-                templateMetadataView.pendingSave = templateMetadata.pendingSave;
-
-                if (!templateMetadataView.hasError) {
-                    templateMetadataView.progressIndicatorStep = templateMetadata.progressIndicatorStep;
-                    _sendMessage('setStep',templateMetadata.progressIndicatorStep);
+            model.getBatchMetadata().onMetadataUpdated.subscribe(function() {
+                var batchMetadataView = component.get('v.batchMetadata');
+                var batchMetadata = model.getBatchMetadata();
+                var headerChanged = Boolean(batchMetadataView.pageHeader !== batchMetadata.pageHeader);
+
+                batchMetadataView.labels = batchMetadata.labels;
+                batchMetadataView.mode = batchMetadata.mode;
+                batchMetadataView.hasError = batchMetadata.hasError;
+                batchMetadataView.errorMessage = batchMetadata.errorMessage;
+                batchMetadataView.pageHeader = batchMetadata.pageHeader;
+                batchMetadataView.pendingSave = batchMetadata.pendingSave;
+
+                if (!batchMetadataView.hasError) {
+                    batchMetadataView.progressIndicatorStep = batchMetadata.progressIndicatorStep;
+                    _sendMessage('setStep',batchMetadata.progressIndicatorStep);
                 } else {
                     component.find('notifLib').showNotice({
                         'variant': 'error',
                         'header': $A.get('$Label.c.PageMessagesError'),
-                        'message': templateMetadataView.errorMessage,
+                        'message': batchMetadataView.errorMessage,
                         closeCallback: function() {
                             //callback action here
                         }
                     });
                 }
 
-                if (templateMetadataView.mode === 'view') {
+                if (batchMetadataView.mode === 'view') {
                     component.set('v.isReadOnly', true);
-                } else if (templateMetadataView.mode === 'create' || templateMetadataView.mode === 'edit') {
+                } else if (batchMetadataView.mode === 'create' || batchMetadataView.mode === 'edit') {
                     component.set('v.isReadOnly', false);
-                    if (templateMetadata.mode === 'edit') {
-                        templateMetadata.labels.batchTemplateHeader = $A.get('$Label.c.bgeBatchTemplateEdit')
```

Comment:
We don't need to set the batchTemplateHeader anymore, since the only reference we had to it was in the .cmp file for the template header.